Разбивка компонента «Конструктор пиццы» на мелкие компоненты

AAA55555 opened the pull request:
---
:mortar_board: [Разбивка компонента «Конструктор пиццы» на мелкие компоненты](https://up.htmlacademy.ru/vue/1/user/566661/tasks/3)

Commit:
module2-task1

Разбил на мелкие компоненты - «Конструктор пиццы» (без drag’n drop)

Commit:
module2-task1

Разбил на мелкие компоненты - «Конструктор пиццы»

enotke commented on src/frontend/src/common/components/ItemCounter.vue:
```diff
@@ -0,0 +1,56 @@
+  },
+  data() {
+    return {
+      currentCount: this.count,
```
Лишнее состояние

enotke commented on src/frontend/src/common/components/ItemCounter.vue:
```diff
@@ -0,0 +1,56 @@
+    >
+      <span class="visually-hidden">{{ textBtnFirst }}</span>
+    </button>
+    <input type="text" name="counter" class="counter__input" v-model="count" />
```
Count это пропс, а их изменять нельзя

enotke commented on src/frontend/src/common/components/ItemCounter.vue:
```diff
@@ -0,0 +1,56 @@
+    <button
+      type="button"
+      class="counter__button counter__button--plus"
+      @click="$emit('plusCount')"
```
Можно добавить максимум 3 одинаковых ингредиента

enotke commented on src/frontend/src/modules/builder/components/BuilderPizzaView.vue:
```diff
@@ -0,0 +1,79 @@
+                :key="ingredient.name"
+                v-if="ingredient.count > 0"
+                class="pizza__filling"
+                :class="`pizza__filling--${ingredient.value}`"
```
Класс изменяется также в зависимости от количества ингредиентов

enotke commented on src/frontend/src/modules/builder/components/BuilderPizzaView.vue:
```diff
@@ -0,0 +1,79 @@
+
+    <div class="content__constructor">
+      <app-drop @drop="$emit('setIngredient', $event.ingredient)">
+        <div class="pizza" :class="`pizza--foundation--${size}-${sauce}`">
```
Модификаторы big, small зависят от типа теста, а не от размеров пиццы

Commit:
module2-task1

Разбивка компонента «Конструктор пиццы» на мелкие компоненты.
Исправил замечания ментора

Commit:
module2-task1

Разбивка компонента «Конструктор пиццы» на мелкие компоненты.
Добавил ограничение на перетаскивание, если ингредиента больше 3

enotke commented on src/frontend/src/common/components/ItemCounter.vue:
```diff
@@ -8,11 +8,18 @@
+      type="text"
+      name="counter"
+      class="counter__input"
+      v-model="count"
```
count это пропс, его нельзя изменять
